pkg/actions/lua/storage/aws/glue.go

```diff
@@ -223,3 +225,76 @@ func getTable(c *GlueClient) lua.Function {
 		return 2
 	}
 }
+
+func createDatabase(c *GlueClient) lua.Function {
+	return func(l *lua.State) int {
+		client := c.client()
+		database := aws.String(lua.CheckString(l, 1))
+
+		var createDBInput *glue.CreateDatabaseInput
+		errorOnAlreadyExists := true
+		if !l.IsNone(2) {
+			lua.CheckType(l, 2, lua.TypeTable)
+			l.Field(2, "error_on_already_exists")
+			l.Field(2, "create_db_input")
+
+			// get flag
+			if !l.IsNil(-2) {
+				errorOnAlreadyExists = l.ToBoolean(-2)
+			}
+
+			// get dbInput
+			if !l.IsNil(-1) {
+				createDBInputJSON := lua.CheckString(l, -1)
+
+				err := json.Unmarshal([]byte(createDBInputJSON), &createDBInput)
+				if err != nil {
+					lua.Errorf(l, "%s", err.Error())
+					panic("unreachable")
+				}
+			}
+		}
+
+		if createDBInput == nil {
+			createDBInput = &glue.CreateDatabaseInput{}
+		}
+		if createDBInput.DatabaseInput == nil {
+			createDBInput.DatabaseInput = &types.DatabaseInput{}
+		}
+		createDBInput.DatabaseInput.Name = database
+
+		// AWS API call
+		_, err := client.CreateDatabase(c.ctx, createDBInput)
+		if err != nil {
+			if !errorOnAlreadyExists {
+				return 0
+			}
+			lua.Errorf(l, "%s", err.Error())
+			panic("unreachable")
+		}
+		return 0
+	}
+}
```

Comment:
I would like to suggest an alternative to this code I find few issues here: 
1. `l.IsNone(2)` misses the case of `nil` should use `l.IsNoneOrNil(2)`
2. The parsing of of the dbOpts is quite hard to read and not intuitive for go lang reader (`l.Field(2`, `!l.IsNil(-2)` etc) 
3. The `json.marshal(...)` is not required and prone to errors if the caller forgets to use it. 
4. The fact that `db` (name) will always take place is weird I'd expect if I set it in the CreateDBInput it will override, I suggest to treat it that way. 

```suggestion
import "github.com/mitchellh/mapstructure"

type glueCreateDatabaseOpts struct {
	CreateDBInput        *glue.CreateDatabaseInput `json:"create_db_input" mapstructure:"create_db_input"`
	ErrorOnAlreadyExists bool                      `json:"error_on_already_exists" mapstructure:"error_on_already_exists"`
}

func createDatabase(c *GlueClient) lua.Function {
	return func(l *lua.State) int {
		client := c.client()
		database := lua.CheckString(l, 1)
		createDbOpts := &glueCreateDatabaseOpts{
			CreateDBInput: &glue.CreateDatabaseInput{
				DatabaseInput: &types.DatabaseInput{
					Name: aws.String(database),
				},
			},
			ErrorOnAlreadyExists: true,
		}
		// decode opts if provided and override defaults
		if !l.IsNoneOrNil(2) {
			var t interface{}
			t, err := util.PullTable(l, 2)
			if err != nil {
				lua.Errorf(l, "PullTable: %s", err.Error())
				panic("unreachable")
			}
			err = mapstructure.Decode(t, createDbOpts)
			if err != nil {
				lua.Errorf(l, "DecodeTable: %s", err.Error())
				panic("unreachable")
			}
		}

		// AWS API call
		_, err := client.CreateDatabase(c.ctx, createDbOpts.CreateDBInput)

		if err != nil {
			existErr := &types.AlreadyExistsException{}
			if errors.As(err, &existErr) && !createDbOpts.ErrorOnAlreadyExists {
				return 0
			}
			lua.Errorf(l, "%s", err.Error())
			panic("unreachable")
		}
		return 0
	}
}
```

Reply:
re. #2, this is the standard way to parse a table when binding to lua- I found lots of examples like this when searching for how to parse a table parameter

re#3, this was the design we agreed upon- the higher level exporter function should take a table, but the lower level crerateDatabase function should take AWS JSON just like getTable and createTable.

re #4 I agree it's a bit weird, but this function does not live in a vacuum, it is part of a Lua/Glue API we expose and wrote it to be consistent with createTable(which you authored, by the way)